soc: nordic: nrf54l: Setup power and clock only on secure build

nordic-krch opened the pull request:
Setup regulators and oscillators only on cpuapp secure target.

nordic-krch committed:
soc: nordic: nrf54l: Setup power and clock only on secure build

Setup regulators and oscillators only on cpuapp secure target.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
soc: nordic: nrf54l: Explicitly include autoconf.h

Include configuration file for cases when this file is complied
in special builds (e.g. TFM).

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

masz-nordic commented on soc/nordic/nrf54l/soc.c:
```diff
@@ -20,38 +23,33 @@
-	sys_cache_instr_enable();
-
+/* NRF_REGULATORS and NRF_OSCILLATORS are configured to be secure
+ * as NRF_REGULATORS.POFCON is needed by the secure domain to
```
```suggestion
 * as NRF_REGULATORS.POFCON is needed by the secure image to
```

nordic-krch committed:
drivers: clock_control: nrf: Move NRF_PERIPH_GET_FREQUENCY

Move macro from nrf_clock_control.h to soc_nrf_common.h. Clock control
header fetches many dependencies (e.g. onoff.h) so move macro to more
low level header.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

aescolar commented on soc/nordic/common/soc_nrf_common.h:
```diff
@@ -236,6 +236,25 @@
+ *
+ * @return Frequency of the clock that is used for the node.
+ */
+#define NRF_PERIPH_GET_FREQUENCY(node) \
```
Due to this move, we'd need to ensure this header is included also for simulated targets.
You could do
```diff
diff --git a/drivers/counter/counter_nrfx_timer.c b/drivers/counter/counter_nrfx_timer.c
index afbdb4ed1ad..6af5a31ad15 100644
--- a/drivers/counter/counter_nrfx_timer.c
+++ b/drivers/counter/counter_nrfx_timer.c
@@ -3,6 +3,7 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
+#include <soc.h>
 #include <zephyr/drivers/counter.h>
 #include <zephyr/drivers/clock_control/nrf_clock_control.h>
 #include <zephyr/devicetree.h>
```
(note this header happens to beincluded thru a quite long path for arm:
```
# 7 "drivers/counter/counter_nrfx_timer.c"
# 14 "include/zephyr/drivers/clock_control/nrf_clock_control.h"
# 11 "include/zephyr/sys/onoff.h"
# 17 "include/zephyr/kernel.h"
# 36 "include/zephyr/kernel_includes.h"
# 19 "include/zephyr/arch/cpu.h"
# 30 "include/zephyr/arch/arm/arch.h"
# 18 "include/zephyr/arch/arm/asm_inline.h"
# 24 "include/zephyr/arch/arm/asm_inline_gcc.h"
# 10 "modules/cmsis/./cmsis_core.h"
# 24 "modules/cmsis/./cmsis_core_m.h"
# 14 "soc/nordic/nrf52/./soc.h"
# 1 "soc/nordic/common/./soc_nrf_common.h" 1
```

aescolar reviewed (dismissed):
see previous comment

nordic-krch commented:
@anangl @masz-nordic can you take another look? Even though it might not get into 4.1 we are cherry-picking it to ncs so it would be good to get it approved so it won't diverge.

Vge0rge commented:
This is ready to get merged, right?

aescolar commented:
> This is ready to get merged, right?

@Vge0rge yes, but we are in 4.1 freeze until the end of this week
